examples/EA/SOO/TSP.tpp

```diff
@@ -81,149 +81,144 @@ const double cities[10][10] = {
-
-	typedef Individual< Tour,double > IndividualType;
-	typedef std::vector< IndividualType > Population;
-	
-	bool compare_fitness( const IndividualType & a, const IndividualType & b ) {
-		return( a.unpenalizedFitness() < b.unpenalizedFitness() );
-	}
+typedef boost::adjacency_matrix< boost::undirectedS,
+	boost::property< boost::vertex_color_t, std::string >,
+	boost::property< boost::edge_weight_t, double,
+		boost::property< boost::edge_color_t, std::string >
+	>,
+	boost::no_property 
+> Graph;
+typedef boost::graph_traits<Graph>::vertex_descriptor Vertex;
+typedef boost::graph_traits<Graph>::edge_descriptor Edge;
+typedef boost::property_map<Graph, boost::edge_weight_t>::type WeightMap;
+typedef boost::property_map<Graph, boost::edge_color_t>::type ColorMap;
+
+typedef Individual< Tour,double > IndividualType;
+typedef std::vector< IndividualType > Population;
+
+
+/**
+ * \brief Calculates the cost of a tour w.r.t. to a cost matrix.
+ */
+struct TspTourLength : public SingleObjectiveFunction{
+	typedef SingleObjectiveFunction base_type;
 
 	/**
-	 * \brief Calculates the cost of a tour w.r.t. to a cost matrix.
+	 * \brief Default c'tor, initializes the cost matrix.
 	 */
-	struct TspTourLength : public shark::SingleObjectiveFunction{
-		typedef shark::SingleObjectiveFunction base_type;
-
-		/**
-		 * \brief Default c'tor, initializes the cost matrix.
-		 */
-		TspTourLength( const shark::RealMatrix & costMatrix = shark::RealMatrix() ) : m_costMatrix( costMatrix )
-		{ }
-
-		std::string name() const
-		{ return "TspTourLength"; }
-
-		std::size_t numberOfVariables() const
-		{ return m_costMatrix.size1(); }
-
-		/**
-		 * \brief Calculates the costs of the supplied tour.
-		 */
-		base_type::ResultType eval( const base_type::SearchPointType & input ) const {		    	    
-			SIZE_CHECK( input.size() == m_costMatrix.size1() );
-		    m_evaluationCounter++;
-			base_type::ResultType result( 0 );
-			for( std::size_t i = 0; i < input.size() - 1; i++ ) {
-				result += m_costMatrix( input( i ), input( i+1 ) );
-			}
-		    return( result );
+	TspTourLength( const RealMatrix & costMatrix = RealMatrix() ) : m_costMatrix( costMatrix )
+	{ }
+
+	std::string name() const
+	{ return "TspTourLength"; }
+
+	std::size_t numberOfVariables() const
+	{ return m_costMatrix.size1(); }
+
+	/**
+	 * \brief Calculates the costs of the supplied tour.
+	 */
+	base_type::ResultType eval( const base_type::SearchPointType & input ) const {		    	    
+		SIZE_CHECK( input.size() == m_costMatrix.size1() );
+	    m_evaluationCounter++;
+		base_type::ResultType result( 0 );
+		for( std::size_t i = 0; i < input.size() - 1; i++ ) {
+			result += m_costMatrix( input( i ), input( i+1 ) );
 		}
+	    return( result );
+	}
 
-		shark::RealMatrix m_costMatrix;
-	};
+	RealMatrix m_costMatrix;
+};
+
+/**
+ * @brief Implements partially mapped crossover
+ * 
+ * Makes sure that only correct tours on the graph
+ * are generated.
+ */
+struct PartiallyMappedCrossover {

```

Comment:
I did not even know that we have an implementation of PMX. This should move into its own file in Operators/Recombination/PMX.h